### PR TITLE
Spec updates related to UIC

### PR DESCRIPTION
**Pull request by drench**
This updates the `UsersController` specs to use the `create_users` flag to determine if `User#edit` and `User#update` are available, and changes the `Product` model specs to not rely on user-based price groups, which are not available in the UIC fork.

This change should resolve the spec failures in https://github.com/tablexi/nucore-uic/pull/113 when merged.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -55,8 +55,8 @@
       @params[:id] = user.id
     end
 
-    it_should_allow_admin_only do
-      expect(assigns[:user]).to eq(user)
+    context "when create_users is active", feature_setting: { create_users: true } do
```
Comment (meara) [inline spec/controllers/users_controller_spec.rb]: This seems reasonable to me, but I recall someone mentioning that with feature-flags like this, we prefer to test both contexts (with the feature on and the feature off).

What makes this case different?

Comment (drench) [inline spec/controllers/users_controller_spec.rb]: In general that's how I'd do it. In this case, there was a set of existing specs below starting at line 308 that turn the `create_users` feature off and test that the routes don't exist, so I added `#edit` and `#update` there.

Comment (jhanggi) [inline spec/controllers/users_controller_spec.rb]: I would probably pull this flag up to the `describe "GET #edit"` line, but it's not a big deal here. But in general, if having the feature off means there is no behavior, it's usually safe to not do the inverse.

Comment (meara) [inline spec/controllers/users_controller_spec.rb]: Thanks folks!

**Pull request merged**